### PR TITLE
Feature/airtable

### PR DESCRIPTION
**Pull request by iruzevic**
### Added
- New `dynamic` block and component to allow dynamic form fields.
- Integration form picker now supports clear option.
- Airtable integration now supports dynamic connected tables.
- Airtable integration now supports multiple select fields.

### Fixed
- Integration sync fix when you unset attributes that are set in manifest as default.

### CODE REVIEW

File: src/Blocks/components/dynamic/components/dynamic-editor.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		additionalFieldClass,
+		blockClientId,
+		additionalClass,
+	} = attributes;
+
+	const dynamicName = checkAttr('dynamicName', attributes, manifest);
+	const dynamicCustomLabel = checkAttr('dynamicCustomLabel', attributes, manifest);
+
+	const dynamicClass = classnames([
+		selector(componentClass, componentClass),
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/dynamic/components/dynamic-editor.js:33]: Please use `classnames` from Libs

File: src/Blocks/components/dynamic/components/dynamic-options.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	const dynamicIsRequired = checkAttr('dynamicIsRequired', attributes, manifest);
+	const dynamicTracking = checkAttr('dynamicTracking', attributes, manifest);
+	const dynamicDisabledOptions = checkAttr('dynamicDisabledOptions', attributes, manifest);
+	const dynamicIsMultiple = checkAttr('dynamicIsMultiple', attributes, manifest);
+
+	return (
+		<PanelBody title={title}>
+			<IconToggle
+				icon={icons.cursorDisabled}
+				label={__('Is Deactivated', 'eightshift-forms')}
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/dynamic/components/dynamic-options.js:42]:
```suggestion
				label={__('Deactivated', 'eightshift-forms')}
```

File: src/Blocks/components/dynamic/components/dynamic-options.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+			{!dynamicIsDeactivated &&
+				<>
+					<Section icon={icons.options} label={__('General', 'eightshift-forms')}>
+						<NameField
+							value={dynamicName}
+							attribute={getAttrKey('dynamicName', attributes, manifest)}
+							disabledOptions={dynamicDisabledOptions}
+							setAttributes={setAttributes}
+							type={'dynamic'}
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/dynamic/components/dynamic-options.js:57]:
```suggestion
							type='dynamic'
```

File: src/Blocks/components/integrations/components/integrations-options.js [hunk 4]
```diff
@@ -159,18 +159,26 @@ export const IntegrationsOptions = ({
...
+								} else {
+									updateIntegrationBlocks(clientId, postId, block, value.toString());
+									setAttributes({ [itemIdKey]: value.toString() });
+								}
 							}
 						}}
 						reducedBottomSpacing={innerIdKey && itemId}
 						closeMenuAfterSelect
 						simpleValue
+						clearable={true}
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/integrations/components/integrations-options.js:181]:
```suggestion
						clearable
```

File: src/Blocks/components/integrations/components/integrations-options.js [hunk 5]
```diff
@@ -179,11 +187,17 @@ export const IntegrationsOptions = ({
...
+								if (!value) {
+									setAttributes({ [innerIdKey]: undefined });
+								} else {
+									updateIntegrationBlocks(clientId, postId, block, itemId, value.toString());
+									setAttributes({ [innerIdKey]: value.toString() });
+								}
 							}}
 							closeMenuAfterSelect
 							simpleValue
+							clearable={true}
```
Comment (goranalkovic-infinum) [inline src/Blocks/components/integrations/components/integrations-options.js:200]:
```suggestion
							clearable
```

File: src/Integrations/Airtable/Airtable.php [hunk 6]
```diff
@@ -300,6 +286,35 @@ function ($checkbox) {
...
+					$output[] = [
+						'component' => 'dynamic',
+						'dynamicType' => 'select',
+						'dynamicName' => $name,
+						'dynamicTracking' => $name,
+						'dynamicIsDeactivated' => true,
+						'dynamicFieldLabel' => $label,
+						'dynamicTypeCustom' => 'dynamicSelect',
+						// translators: %1$s will be replaced with field label and %2$s with field name.
+						'dynamicCustomLabel' => \sprintf(\__('We will display dinamic data on the frontend for %1$s field - %2$s.', 'eightshift-forms'), $label, $name),
```
Comment (goranalkovic-infinum) [inline src/Integrations/Airtable/Airtable.php:310]:
```suggestion
						'dynamicCustomLabel' => \sprintf(\__('We will display dynamic data on the frontend for %1$s field - %2$s.', 'eightshift-forms'), $label, $name),
```

**Review by goranalkovic-infinum — CHANGES_REQUESTED** [review]
Toeto

**Review by kancijan — APPROVED** [review]

**Review by kancijan — APPROVED** [review]

**Review by goranalkovic-infinum — APPROVED** [review]

**Pull request merged**